operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,267 @@
+  float point_x2 = box[1];
+  float point_x3 = box[2];
+  float point_x4 = box[3];
+  if (bbox_mode == BoundingBoxFormat::Center_XYWH) {
+    point_x1 = box[0] - point_x3 / 2;
+    point_x2 = box[1] - point_x4 / 2;
+    point_x3 = box[0] + point_x3 / 2;
+    point_x4 = box[1] + point_x4 / 2;
+  } else if (bbox_mode == BoundingBoxFormat::XYWH) {
+    point_x1 = box[0];
+    point_x2 = box[1];
+    point_x3 = box[0] + box[2];
+    point_x4 = box[1] + box[3];
+  }
+  int64_t x_start = static_cast<int64_t>(std::round(point_x1));
+  int64_t y_start = static_cast<int64_t>(std::round(point_x2));
+  int64_t x_end = static_cast<int64_t>(std::round(point_x3));
+  int64_t y_end = static_cast<int64_t>(std::round(point_x4));
+
+  auto offset = thickness / 2;
+  x_start -= offset;
+  y_start -= offset;
+  x_end += (thickness - offset);
+  y_end += (thickness - offset);
+
+  x_start = std::max<int64_t>(x_start, 0L);
+  y_start = std::max<int64_t>(y_start, 0L);
+  x_end = std::min<int64_t>(x_end, image.height - 1);
+  y_end = std::min<int64_t>(y_end, image.width - 1);
+
+  thickness = std::clamp<int64_t>(thickness, 1, std::min(x_end - x_start, y_end - y_start));
+  if (x_end - x_start < thickness || y_end - y_start < thickness) {
+    ORTX_CXX_API_THROW("box is too small to draw", ORT_INVALID_ARGUMENT);
+    return;
+  }
+
+  if (x_start + thickness >= image.height || y_start + thickness >= image.width) {
+    ORTX_CXX_API_THROW("box is too small to draw", ORT_INVALID_ARGUMENT);
+    return;
+  }
+
+  // line  (1) --
+  DrawLineInHorizon(image, x_start, y_start, y_end - y_start, color, thickness);
+
+  // line  (2) |--
+  DrawLineInVertical(image, x_start, y_start, x_end - x_start, color, thickness);
+
+  // line  (3) __
+  DrawLineInHorizon(image, x_end - thickness, y_start, y_end - y_start, color, thickness);
+
+  // line  (4) --|
+  DrawLineInVertical(image, x_start, y_end - thickness, x_end - x_start, color, thickness);
+}
+
+void DrawBoxesForNumClasses(ImageView& image, const BoxArray& boxes, int64_t thickness) {
+  std::unordered_map<float, size_t> color_used;
+  std::vector<std::pair<size_t, int64_t>> box_reverse;
+  box_reverse.reserve(boxes.NumBoxes());
+  for (size_t i = 0; i < boxes.NumBoxes(); ++i) {
+    const auto box = boxes.GetBox(i);
+    if (color_used.find(box[kBoxClassIndex]) == color_used.end()) {
+      if (color_used.size() >= KBGRColorMap.size()) {
+        // "The number of classes is larger than the number of colors in the color map.";
+        continue;
+      }
+      color_used.emplace(box[kBoxClassIndex], color_used.size());
+    }
+    box_reverse.emplace_back(i, static_cast<int64_t>(color_used[box[kBoxClassIndex]]));
+  }
+
+  // A class which has higher score will be drawn on the top of the image.
+  std::sort(box_reverse.begin(), box_reverse.end(),
+            [](const std::pair<size_t, int64_t>& first_, const std::pair<size_t, int64_t>& second_) {
+              return first_.second < second_.second;
+            });
```

Comment:
Can we use a reverse iterator instead of sorting?

Reply:
We are sorting this by colors, because smaller color index has higher priority.  And the `box_reverse` is sorted by score, so this is essential.